src/file_watcher/mock.rs

```diff
@@ -0,0 +1,55 @@
+use std::io::{Error, ErrorKind, Result};
+use std::path::PathBuf;
+use std::vec::Vec;
+
+pub struct FileWatcherImpl {
+    highest_id: usize,
+    watches: Vec<FileWatchImpl>
+}
+
+pub struct FileWatchImpl {
+    id: usize,
+}
+
+impl FileWatcherImpl {
+    pub fn init() -> Result<FileWatcherImpl>  {
+        return Result::Ok(FileWatcherImpl {
+            highest_id: 0,
+            watches: vec![]
+        });
+    }
+
+    pub fn add_watch(&mut self, _file_path: &PathBuf) -> Result<&FileWatchImpl> {
+        let fw = FileWatchImpl {
+            id: self.highest_id,
```

Comment:
I think the better question is - do we expect to actually *need* to be able to remove watches? Because currently it seems that we observe the script+input files and keep watching them until the program exits. As such, we could get rid of the `rm_watch()` part of the interface, which in turn would mean that `add_watch()` can simply return `Result<()>`.

Reply:
Does it work on BSD as of now without removing watches on reload? I’m struggling testing it a bit. Otherwise, I think you’re right and I don’t see use case for removing switches. I don’t plan building an IDE from this, so it should be fine to remove them if they are not needed for this basic functionality.

Reply:
Ok, so I did some testing and either I'm using the kqueue crate wrong, or something's broken, since I wrote a C program using `kqueue()` and `kqevent()` and it worked fine.

Though I did find out that there's one thing we need to change, that being adding `NOTE_RENAME` and `NOTE_DELETE` to the flags, since otherwise we miss on changes that are done in the "create new file, write, replace old file with new one" fashion.

Reply:
I poked at the kqueue create wrong and interestingly, things work fine if I remove one of the file watches (no matter if it's the script watch, or the input file watch) - when the program watches only a single file, the events arrive as expected and the program doesn't die with an OS error when reloading.

Reply:
Uhhh, the kqueue crate is 700-line long single file, which doesn't give too much confidence. It's certainly possible it's broken. I'll try creating an issue if the project is still alive and maintained. I'll try to help with debugging as soon as I get the BSD finally working as expected.

Btw great thing you noticed the `NOTE_RENAME` and `NOTE_DELETE`, I missed it and didn't even think about it.